op-program/client/boot.go

```diff
@@ -45,22 +55,39 @@ func (br *BootstrapClient) BootInfo() *BootInfo {
 	l2OutputRoot := common.BytesToHash(br.r.Get(L2OutputRootLocalIndex))
 	l2Claim := common.BytesToHash(br.r.Get(L2ClaimLocalIndex))
 	l2ClaimBlockNumber := binary.BigEndian.Uint64(br.r.Get(L2ClaimBlockNumberLocalIndex))
-	l2ChainConfig := new(params.ChainConfig)
-	err := json.Unmarshal(br.r.Get(L2ChainConfigLocalIndex), &l2ChainConfig)
-	if err != nil {
-		panic("failed to bootstrap l2ChainConfig")
-	}
-	rollupConfig := new(rollup.Config)
-	err = json.Unmarshal(br.r.Get(RollupConfigLocalIndex), rollupConfig)
-	if err != nil {
-		panic("failed to bootstrap rollup config")
+	l2ChainID := binary.BigEndian.Uint64(br.r.Get(L2ChainIDLocalIndex))
+
+	var l2ChainConfig *params.ChainConfig
+	var rollupConfig *rollup.Config
+	if l2ChainID == CustomChainIDIndicator {
```

Comment:
`CustomChainIDIndicator` allows users to configure custom chains that have the same chain ID as a preset. Users that have such configs should feel bad but honestly I don't see a compelling reason to prohibit that usecase.